sympy/physics/mechanics/kane.py

```diff
@@ -343,44 +395,40 @@ def _initialize_constraint_matrices(self, config, vel, acc):
             self._A_rs = Matrix()
 
     def _initialize_kindiffeq_matrices(self, kdeqs):
-        """Initialize the kinematic differential equation matrices."""
+        """Initialize the kinematic differential equation matrices.
+
+        The kinematic differential equations are in this form::
+
+            K_ku(q, t) * u + K_kqdot(q, t) * q' + f_k(q, t) = 0
+
+        This method finds and stores K_ku, K_kqdot, and f_k. It also solves the
+        equations for q' and stores the solution.
+
+        """
 
         if kdeqs:
-            if len(kdeqs) != len(self.num_coordinates):
+            if len(kdeqs) != self.num_coordinates:
                 msg = ('There must be an equal number of kinematic '
                        'differential equations and generalized coordinates.')
                 raise ValueError(msg)
 
             kdeqs = Matrix(kdeqs)
 
-            u = self.u
-            qdot = self._qdot
-
             # Substitution dictionaries for setting variables to zero.
-            u_zero = dict((i, 0) for i in u)
+            u_zero = dict((i, 0) for i in self.u)
```

Comment:
I noticed when looking through a few weeks ago that all of the internal code in `KaneMethod` would use the private attribute over the public attribute with the only exceptions being u and q (though this may no longer be the case with the large volume of changes propose). I was wondering if using the private attributes would be better practice for the internal code or if there's no real stylistic difference.